### PR TITLE
Font Library: adding docs

### PR DESCRIPTION
**Pull request by matiasbenedetto**
:warning: this  is a work in progress
Some of the things documented here can change soon.

## What?
Font Library: adding docs

## Why?
To inform the key concepts and functionality around Font Library.

## How?
By documenting the existing functionality.

## Testing Instructions
- Read the docs.
- Check if they make sense.
- Propose changes and additions.

### CODE REVIEW

File: lib/experimental/fonts/font-library/README.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+- **Install font family**: Is to make a font family available to the user in the Font Library. The user will be able to activate and use the installed font families.
+- **Activate font family**: Is to make it ready to use in the site or post editor. All the active fonts will appear in the font pickers so the user can use them in the site elements.
+- **Deactivate font family**: Is to make a font family unusable. If a font family is not active, it won't appear in the font pickers.
+- **Uninstall font family**: Is to remove the font family from the Font Library permanently. 
+
+### Different types of installations
+
+The library allows extenders to define how WordPress will install fonts. When users install a font family, its definition will always be saved to the database. What can vary is where are the font face file assets stored.
+
+Let's see the different types of installations:
```
Comment (annezazu) [inline lib/experimental/fonts/font-library/README.md]: The "Let's see" and "Let's say" language is a bit too casual. I'd recommend cleaning it up a bit: "Please review the following different types of installations to find what works best for your use case" and "In a situation where you want to install a font family that doesn't need any font assets to work, these fonts are usually called system fonts. You can define these font families like this:"

File: lib/experimental/fonts/font-library/README.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+          "fontWeight": "400",
+          "fontStyle": "normal",
+          "fontFamily": "Piazzolla"
+        },
+    ]
+}
+```
+
+### Extensibility
+Font Library is able to manage different font collections. A font collection is simply a list of font families ready to be installed by the user. Extenders can provide multiple typographic collections.
```
Comment (annezazu) [inline lib/experimental/fonts/font-library/README.md]: I'd add a note here that any font collection will be available as a new tab in the font library modal so folks know what to expect.

File: lib/experimental/fonts/font-library/README.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    }
+  ],
+  "categories": [
+    "sans-serif",
+    "serif"
+  ]
+}
+```
+
+### Create a plugin to provide my font collection
```
Comment (annezazu) [inline lib/experimental/fonts/font-library/README.md]: "Create a plugin to provide a font collection"

File: lib/experimental/fonts/font-library/README.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+Creating a plugin to provide a font collection can be as simple as this:
```
Comment (annezazu) [inline lib/experimental/fonts/font-library/README.md]: I would rephrase this to: "Here's an example of a plugin that provides a font collection:"

File: lib/experimental/fonts/font-library/README.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+Save a font family may or may not include storing the font files assets localy to disk in `/wp-content/fonts` folder. That depends on how the font family sent to this endpoint is defined.
+
+```
+POST /wp-json/wp/v2/fonts
+```
+
+Example:
+
+#### Uninstall font families
+Uninstall a font family is to remove the font definition from WordPress and make it unavailable for users to activate and save.
```
Comment (annezazu) [inline lib/experimental/fonts/font-library/README.md]: This is a bit hard to read through. I would rephrase to "Uninstalling a font family removes the font definition from WordPress and makes it unavailable for users to activate and save."

**Review by annezazu — COMMENTED** [review]
In case it helps, left some quick copy feedback :)

File: lib/experimental/fonts/font-library/README.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+          "fontStyle": "normal",
+          "fontFamily": "Piazzolla"
+        },
+    ]
+}
+```
+
+####  Install a font from external sources and serve it from your site
+Let's say you want to install a font family that uses assets located outside of your site and download it while installing to serve them always from your WordPress `/wp-content/fonts` folder. This can be useful if you want to avoid depending on external sites for technical or legal reasons. In that case, you need to set the `downloadFromUrl` property on each font face.
+
```
Comment (vcanales) [inline lib/experimental/fonts/font-library/README.md]: We'll need to change this once the ability to change this directory is implemented, in order to specify that the directory might not always be `/wp-content/fonts`.

File: lib/experimental/fonts/font-library/README.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+          "fontFamily": "Piazzolla"
+        },
+    ]
+}
+```
+
+#### Install a font providing the font asset
+Let's say you want to install a font family using the font files you are providing in an HTTP request. These font file assets will be stored in your WordPress `/wp-content/fonts` folder and they will always be served from your site. As the previous way, this can be useful if you want to avoid depending on external sites for technical or legal reasons. You need to add the file to your HTTP request and add a reference for it in the font face definition.
+
+
```
Comment (vcanales) [inline lib/experimental/fonts/font-library/README.md]: Same as above; this might not always be the directory where fonts are served from in the future.

**Comment by mikachan** [comment]
Noting that we should find a good place in the docs to put the information from https://github.com/WordPress/gutenberg/issues/57980.

**Comment by colorful-tones** [comment]
We'll want to add a section on how to deactivate the Font Library with filter, see #57818 

Here is the code copied from merged ticket:

```
function disable_font_library_ui( $editor_settings ) { 
    $editor_settings['fontLibraryEnabled'] = false;
    return $editor_settings; 
}

add_filter( "block_editor_settings_all", "disable_font_library_ui" );
```

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+        "preview": "https://s.w.org/images/fonts/17.6/previews/abril-fatface/abril-fatface.svg"
+      },
+      "categories": ["serif"]
+    },
+  ]
+}
+```
+
+#### Unregister a font collection
+To unregister a font collection you need to add an action calling `wp_unregister_font_collection` with the font collection slug as the only parameter. The unregistered font collections won't appear as a font library modal tab and it won't be listed on the `/font-collections/` REST API endpoint.
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:181]:
```suggestion
To unregister a font collection, add an action calling `wp_unregister_font_collection` with the font collection slug as the only parameter. Unregistered font collections won't appear in the Font Library modal tab list, and won't be returned from the `/font-collections/` REST API endpoint.
```
Massaging this a little.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+
+Example:
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:183]:
```suggestion
Example:

```

Added a line break above code block. To my eyes this is easier to scan in .md.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+add_action( 'init', 'register_my_collection' );
+```
+
+Example using a JSON from an URL:
+```php
+function register_my_collection(){
+    wp_register_font_collection ( 'my-font-foundry', 'https://example.com/fonts/my-collection.json' );
+}
+
+add_action( 'init',
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:125]:
```suggestion
add_action( 'init', 'register_my_collection' );
```
Fix.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+add_action( 'init', 
+```
+
+`my-collection.json` file example:
+
+```json
+{
+  "$schema": "https://schemas.wp.org/trunk/font-collection.json",
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:132]:
```suggestion
  "$schema": "https://schemas.wp.org/trunk/font-collection.json",
  "slug": "my-font-foundry-collection",
```
Adds `slug`.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+  "name": "My Font Foundry Collection",
+  "decription": "The font offered by My Font Foundry",
+  "categories": [
+    { "id": "sans-serif", "name": "Sans Serif" },
+    { "id": "serif", "name": "Serif" },
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:137]:
```suggestion
    {
      "name": "Sans Serif",
      "slug": "sans-serif",
    },
    {
      "name": "Serif",
      "slug": "serif"
    },
```
Reformat and switch `id` to `slug`.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+Example using a local JSON file:
+```php
+function register_my_collection(){
+    wp_register_font_collection ( 'my-font-foundry', path_join( __DIR__, 'my-collection.json' ) );
+}
+
+add_action( 'init', 'register_my_collection' );
+```
+
+Example using a JSON from an URL:
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:119]:
```suggestion
Example using a JSON from an URL:

```
Adds line break before code block.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+function register_my_collection(){
+    wp_register_font_collection ( 'my-font-foundry', $config );
+}
+
+add_action( 'init', 'register_my_collection' );
+```
+
+##### Register a font collection using a JSON file
+
+Example using a local JSON file:
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:110]:
```suggestion
Example using a local JSON file:

```
Adds line break before code block.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+
+The only responsibility of a font collection is to present a list of font families in a correct format. Font collections are not in charge of installing the fonts; they just list a group of font families so users can choose the one they want to install. If the font family includes font faces, those font faces should have links to bundled (e.g: in a plugin assets) or external (e.g: from a CDN) font assets that will be downloaded or linked when the font family is installed.
+
+#### Register a font collection
+Register a font collection means providing a user a new list of fonts from one can select and install fonts. When a font collection is registered it is featured as a new tab in the font library modal and it's fetcheable using the `/font-collections/` REST API endpoint.
+
+To register a font collection you need to use `wp_register_font_collection` function.
+
+##### Register a font collection using PHP only
+Example:
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:50]:
```suggestion
Example:

```
Adds line break before code block.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+## Rest API
+
+### Different types of installations
+
+The library allows extenders to define how WordPress installs fonts. When users install a font family, its definition will always be saved to the database. What can vary is where the font face file assets are stored.
+
+Review the following different types of installations to find what works best for your use case:
+#### Install a system font ( a font with no font file assets)
+When you want to install a font family that doesn't need any font assets to work, these fonts are usually called system fonts. You can define these font families like this:
+```json
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:206]:
```suggestion

```json
```
Adds line break before code block.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+    "name": "Humanist",
+    "slug": "humanist",
+    "fontFamily": "Seravek, 'Gill Sans Nova', Ubuntu, Calibri, 'DejaVu Sans', source-sans-pro, sans-serif",
+
+}
+```
+
+#### Install a font with external assets
+In a situation where you want to install a font family that needs to add font faces using external font file assets, you can define font families like this:
+```json
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:217]:
```suggestion

```json
```
Adds line break before code block.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+
+```
+GET /wp-json/wp/v2/fonts/collections
+```
+
+Example:
+
+#### Get a font collection
+Get a particular font collection and its data.
+```
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:309]:
```suggestion

```console
```
Adds line break before code block, updates formatting.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+DELETE /wp-json/wp/v2/fonts
+```
+
+Example:
+
+#### Get font collections
+Gets the list of fonts collections available.
+
+```
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:301]:
```suggestion
```console
```
Updates formatting.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+
+Example:
+
+#### Uninstall font families
+Uninstalling a font family removes the font definition from WordPress and makes it unavailable for users to activate and save.
+
+If the font family being uninstalled is using local font file assets, those files will be removed from the `/wp-content/fonts` folder.
+
+```
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:292]:
```suggestion
```console
```
Updates formatting.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 21]
```diff
@@ -0,0 +1,314 @@
...
+
+
+### API endpoints
+
+#### Install font families
+Install a font family is to save a font definition in WordPress and make it available to be activated and used by the user.
+
+Save a font family may or may not include storing the font files assets localy to disk in `/wp-content/fonts` folder. That depends on how the font family sent to this endpoint is defined.
+
+```
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:281]:
```suggestion
```console
```
Updates formatting.

**Review by ironprogrammer — COMMENTED** [review]
@matiasbenedetto I was testing custom collections today, so had a few updates to share. I hope these are helpful as the docs are coming together.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+
+add_action( 'init', 'register_my_collection' );
+```
+
+##### Register a font collection using a JSON file
+
+Example using a local JSON file:
+```php
+function register_my_collection(){
+    wp_register_font_collection ( 'my-font-foundry', path_join( __DIR__, 'my-collection.json' ) );
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:113]:
```suggestion
    wp_register_font_collection (
        'my-font-foundry',
        array(
            'name'          => 'My Font Foundry',
            'font_families' => path_join( __DIR__, 'my-collection.json' )
        )
    );
```
This method changed in https://github.com/WordPress/wordpress-develop/commit/8ced706f7f87a8882d5960b23b669d8e5217d54c.

File: lib/compat/wordpress-6.5/fonts/README.md [hunk 23]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+add_action( 'init', 'register_my_collection' );
+```
+
+Example using a JSON from an URL:
+```php
+function register_my_collection(){
+    wp_register_font_collection ( 'my-font-foundry', 'https://example.com/fonts/my-collection.json' );
```
Comment (ironprogrammer) [inline lib/compat/wordpress-6.5/fonts/README.md:122]:
```suggestion
    wp_register_font_collection (
        'my-font-foundry',
        array(
            'name'          => 'My Font Foundry',
            'font_families' => 'https://example.com/fonts/my-collection.json'
        )
    );
```
Same change as noted above in https://github.com/WordPress/gutenberg/pull/54477/files#r1500148930.
